### ISSUE TITLE
[a11y]: ContainedList [usage examples] - content announced in one stretch

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13950]
### Package

@carbon/react

### Browser

FF

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

JAWS FF

### Description

JAWS announces the content in one stretch

When the focus goes to the mentioned text, JAWS is reading the content in one stretch.
Ex: `List item List item details List item details`

Note - Visually we can differentiate each item as 3 separate items but the screen reader is not

### WCAG 2.1 Violation

1.3.2 Meaningful Sequence 

### Reproduction/example

Record `Number 41` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

When the visually separated text is announced together as one single element, The blind user would not realize that the text is visually separated. And also it makes SR over verbose which can be annoying.

### Recommended fix

Screen reader is expected to identify the text separately how it is visible in each line

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by andreancardona** [comment]
Chatted with @mbgower and the functionality / implementation of this component needs further refinement. Need to discuss with design & dev.

**Comment by andreancardona** [comment]
after talking to Gower here are some notes: 

_From our a11y call it seems: contained list is less read-only than a structured list. So although it looks closer to a plain old list, it is actually closer to an actionable data table, with structured list, despite its multiple columns, being a nearer kin to a list


From that perspective, it seems even more true to me that we should treat them as tables under the covers. The moment you do that, there is no more issue for JAWS users (with even minor experience) in regard to hearing too much information._

Also from Tom: 

_Certainly looks like a table to me

I'd think the title is the caption, and then the first column is the heading and the others are data cells_

**Comment by tay1orjones** [comment]
It looks like we're currently mixing native list elements with aria roles, which might be part of the problem. Based on [this article](https://www.scottohara.me/blog/2018/05/26/aria-lists.html), there's some differences in how list and list items are read when using native `ul`, `ol`, `li`, etc. vs using `role="list"` and `role="listitem"`

Right now ContainedList uses the native `ul` and `li` elements, but _also_ places `role="list"` on the `ul`. If we remove the `role="list"` does it help screenreaders separate out the content based on what's focused?

Alternatively, if we keep the `role="list"`, there isn't a `role="listitem"` on the `li` currently. If `role="listitem"` is added to the `li`'s, does it help?

**Comment by dabrad26** [comment]
Any update on this ticket? Or some of the other accessibility tickets that are opened? We have some gov customers reporting some concerns around some of these issues.

**Comment by mbgower** [comment]
@tay1orjones I did a quick check and the answer to these two questions appears to be "no"
> Right now ContainedList uses the native ul and li elements, but also places role="list" on the ul. If we remove the role="list" does it help screenreaders separate out the content based on what's focused?

> Alternatively, if we keep the role="list", there isn't a role="listitem" on the li currently. If role="listitem" is added to the li's, does it help?

**Comment by heloiselui** [comment]
Checked with our accessibility team, and this issue still exists.
